pkg/upgrade/config/v1alpha1/config.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/upgrade/v1alpha1"
+)
+
+// UpgradeConfig is wrapper over apis.UpgradeConfig which is
+// upgrade config for a particular job.
+type UpgradeConfig struct {
+	object *apis.UpgradeConfig
+}
+
+// Builder helps to build UpgradeConfig instance
+type Builder struct {
+	*UpgradeConfig
```

Comment:
Let us use a field for *Config. Let us not embed this. This will help us follow a standard idiomatic maya pattern.

Reply:
added